aten/src/ATen/ThreadLocalState.h

```diff
@@ -87,12 +87,12 @@ class TORCH_API ThreadLocalStateGuard {
 };
 
 template <typename T>
-std::function<T(void)> wrapPropagateTLSState(
-    std::function<T(void)> callback) {
-  return [tls_state = ThreadLocalState(), callback = std::move(callback)]() {
+auto wrapPropagateTLSState(T callback) {
+  return [tls_state = ThreadLocalState(),
+          callback = std::move(callback)](auto&&... args) {
```

Comment:
Thanks for this change! When I added this in https://github.com/pytorch/pytorch/pull/40326 I spent some amount of time trying to make it take variadic args but couldn't quite figure out how the argument forwarding works. I decided not to pursue it further at the time since we didn't really have multiple-arg callbacks in C++ future.